extension/apikeyauthextension/config.go

```diff
@@ -99,3 +117,57 @@ func (cfg *CacheConfig) Validate() error {
 	}
 	return nil
 }
+
+// Validate validates the DynamicResource configuration.
+func (dr *DynamicResource) Validate() error {
+	if dr.Metadata == "" {
+		return fmt.Errorf("metadata must be non-empty")
+	}
+	format := dr.Format
+	if format == "" {
+		format = "%s"
+	}
```

Comment:
A bit odd to define the default on `validate`. Maybe change this to the `createDefault` config function?

Reply:
I've added an Unmarshal method which sets the default now.